### PR TITLE
Add abstain vote functionality

### PR DESCRIPTION
**Pull request by reeved**
# Description
Allows players to abstain from voting during a trial. Abstains negate Votes against the player on trial so if numAbstain >= numVotes, then the player survives.

# Demo

Before: 

After: 
image

# Type of change:
* New feature (Non-breaking change that adds functionality)

# Checklist:
* Have you merged main into your branch?
* Have you tested your changes to ensure it works as expected and does not break existing functionality?
* If applicable, please ensure sufficient tests are added that is related to the changes.
* Please ensure this PR has a `label`, is linked to an `issue` and is related to a `project`
* If new documentation is required for this change, have you created a new documentation issue that describes the documentation needed?

### CODE REVIEW

**Review by Dhors — APPROVED** [review]
Looks good, 
I think it would be great if we could css the button abit more, right now it feels a bit out of place for me.

**Review by ateagit — APPROVED** [review]
LGTM

**Review by parfei — APPROVED** [review]
Nice job :) This is really picky, but imo I think abstain means to not take part of the voting at all (so the timer runs out and you didn't pick anything), but this is more of a "no confidence" vote or something similar, where you vote that no one is actually guilty

File: backend/Events/VoteEvents.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ function voteTrial(io, socket, mafiaGame) {
         const room = mafiaGame.gameRoomsDict[socket.player.roomID];
         const voter = socket.player;
         const votee = voteForDTO.votingFor;
-        room.voteHandler.trialVoteMap[voter.nickname] = room.getPlayerByNickname(votee);
+        room.voteHandler.trialVoteMap[voter.nickname] = votee === 'abstain Vote' ? 'abstain Vote' : room.getPlayerByNickname(votee);
```
[the remark this answers is not in the document]
Comment (reeved) [reply backend/Events/VoteEvents.js:36]: yeah thats a good idea!! this should defs get fixed in a future update.

**Pull request merged**